torch/_inductor/fx_passes/overlap_manual_scheduling.py

```diff
@@ -0,0 +1,583 @@
+            self.data.append(data)
+            self.unique_nodes.add(data)
+
+    def get_child(
+        self, module_stack: str, klass: Optional[type[Any]] = None
+    ) -> Container:
+        if module_stack not in self.children:
+            new_stack = Container(module_stack, klass or self.klass)
+            self.children[module_stack] = new_stack
+        return self.children[module_stack]
+
+    def __getitem__(self, name: str) -> Container:
+        return self.children[name]
+
+    def __getattr__(self, name: str) -> Container:
+        return self.children[name]
+
+    def __repr__(self) -> str:
+        child_lines: list[str] = []
+        for name, child in self.children.items():
+            mod_str = repr(child)
+            mod_str = _addindent(mod_str, 2)
+            child_lines.append(f"({name}): {mod_str}")
+        main_str = f"{self.klass.__name__}("
+        if child_lines:
+            main_str += "\n  " + "\n  ".join(child_lines) + "\n"
+        main_str += ")"
+        return main_str
+
+    def graph_view(self) -> fx.Graph:
+        return _make_subgraph(self.data)
+
+
+def _clean_stack_name(val: str) -> str:
+    name: str = (
+        val.replace("L['self']", "Model")
+        .replace("_modules['", "")
+        .replace("['", ".")
+        .replace("']", "")
+    )
+    return name
+
+
+def _find_key_nodes(nodes: list[fx.Node]) -> tuple[list[fx.Node], list[fx.Node]]:
+    root = []
+    outputs = []
+    nodes_set = OrderedSet(nodes)
+    for node in nodes:
+        for x in node.all_input_nodes:
+            if x not in nodes_set:
+                root.append(x)
+        if all(x not in nodes_set for x in node.users):
+            outputs.append(node)
+    return root, outputs
+
+
+def _make_subgraph(nodes: list[fx.Node]) -> fx.Graph:
+    placeholders, outputs = _find_key_nodes(nodes)
+
+    new_graph = torch.fx.Graph()
+    env: dict[fx.Node, fx.Node] = {}
+
+    def env_lookup(x: fx.Node) -> fx.Node:
+        assert x in env, f"Dependent node {x} not in env when creating downstream node"
+        return env[x]
+
+    def node_copy(
+        node: fx.Node, arg_transform: Callable[[fx.Node], fx.Node]
+    ) -> fx.Node:
+        if node not in env:
+            new_node = new_graph.node_copy(node, arg_transform=arg_transform)
+            env[node] = new_node
+        else:
+            new_node = env[node]
+        return new_node
+
+    for node in placeholders:
+        env[node] = new_graph.placeholder(node.name)
+
+    for node in nodes:
+        if node in placeholders:
+            continue
+        else:
+            new_node = node_copy(node, env_lookup)
+            new_node.meta = node.meta.copy()
+
+    out_node = [env[x] for x in outputs]
+    new_graph.output(out_node)
+    return new_graph
+
+
+def _is_root(stack: str) -> bool:
+    return stack == ""
+
+
+def make_graph_view(graph: fx.Graph) -> Container:
+    """
+    Code from: https://github.com/meta-pytorch/autoparallel/pull/158
```

Comment:
If we're putting this in core, let's do it for real. Dedicated module, docs, tests. cc @fmassa @eellison. I haven't used this API yet so I don't have a real world opinion on it.